scico/linop/_linop.py

```diff
@@ -290,3 +291,41 @@ def __init__(
 
     def _eval(self, x: JaxArray) -> JaxArray:
         return snp.sum(x, axis=self.sum_axis)
+
+
+class Slice(LinearOperator):
+    """A linear operator for slicing an array."""
```

Comment:
or block array, I wonder? I think you could use one of these to select a block from a block array, but probably not use one to select a block and index within the block, if I'm reading the [docs](https://scico.readthedocs.io/en/latest/_autosummary/scico.blockarray.html#indexed-updating) correctly.

(not saying we should add the functionality, but maybe just address this in the docstring)

Reply:
I think that would be worth adding. Let's discuss details.

Reply:
The syntax is already established (first axis indexes the block), but this is not supported due to limitations of the relevant `BlockArray` code:

https://github.com/lanl/scico/blob/445412ef1c4bfb017628f13b35cf8972e51b872f/scico/blockarray.py#L799-L807

As you can see, only `int` or `Ellipsis` indexing expressions are allowed, and there is no support for concatenated indexing of block and block content. It's not clear how difficult it would be to remove the former constraint (one complication would be that indexing could return another `BlockArray` rather than a `JaxArray`), but I think it would be straightforward to add support for the latter.

Reply:
Added support for more general indexing of BlockArray, and documented BlockArray support by Slice.